lib/core/io.js

```diff
@@ -225,8 +224,8 @@ class FetchReader extends Reader {
 	async init() {
 		super.init();
 		if (isHttpFamily(this.url) && !this.preventHeadRequest) {
-			const response = await sendFetchRequest(HTTP_METHOD_HEAD, this.url, this.options);
-			this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_LENGTH));
+			const response = await sendFetchRequest(HTTP_METHOD_GET, this.url, this.options, { range: "bytes=0-0"});
```

Comment:
The implementation makes the flag `preventHeadRequest` obsolete. However, it might be useful to use a HEAD request the get the content length. So, the new implementation should be added to the existing code (instead of replacing the implementation) and activated via another flag, for example.

Reply:
Done.